socorro/external/boto/connection_context.py

```diff
@@ -2,38 +2,72 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-import contextlib
-import datetime
-import json
-import socket
+import io
```

Comment:
This file is based on the `S3Connection` from Antenna. It has the following differences:

* uses configman rather than everett
* drops the health check code
* adds a `load_file` function
* uses different wait times for retrying (Antenna needs to ensure the crash gets to the destination, the processor can just fail and try again later)
* includes `S3Connection` specific tests

Probably best to read this file on its own and not as a diff.